File: swift/RunningSumOf1dArray.swift
// https://leetcode.com/problems/running-sum-of-1d-array/

class Solution {
    func runningSum(_ nums: [Int]) -> [Int] {
        var result = [Int]()
        for (i, num) in nums.enumerated() {
            if i > 0 {
                result.append(num + result[i - 1])
            } else {
                result.append(num)
            }
        }
        return result
    }
}

File: c/RunningSumOf1dArray.c
// https://leetcode.com/problems/running-sum-of-1d-array/

int* runningSum(int* nums, int numsSize, int* returnSize){
    int *returnNums = (int*)malloc(numsSize * sizeof(int));
    returnNums[0] = nums[0];
    
    for (int i = 1; i < numsSize; ++i) {
        returnNums[i] = returnNums[i - 1] + nums[i];
    }
    
    *returnSize = numsSize;
    return returnNums;
}